composer/callbacks/checkpoint_saver.py

```diff
@@ -296,9 +296,9 @@ def __init__(
         latest_artifact_name: Optional[str] = '{run_name}/checkpoints/latest-rank{rank}',
         checkpoint_save_interval: Union[Time, str, int, Callable[[State, Event], bool]] = '1ep',
         *,
-        overwrite: bool = False,
+        overwrite_checkpoints: bool = False,
         num_checkpoints_to_keep: int = -1,
-        weights_only: bool = False,
+        save_model_weights_only: bool = False,
```

Comment:
Add to list of arguments we want to discuss length of variable name on? 

@hanlint I think the right solution here is to revisit our style guide next design meeting and ensure we all are consistent and agree on the vision.